### DISCUSSION TITLE
Reconsidering Demographics Brackets

### DISCUSSION THREAD

**Discussion by SachaG** [Devographics/surveys#181] [General Survey Design]
While we have decided to keep using brackets for most questions, the fact that we're starting a new survey and don't have to worry about legacy data gives us a chance to reconsider the brackets we've been using. 

I think there are three factors to consider:

### 1. Simpler Brackets

@LeaVerou proposed narrowing down the company (or organization) size brackets to:

- Just me!
- Small (2 - 50)
- Medium (51 - 200)
- Large (201 - 1000)
- Very Large (1000+)

This would be good because it makes the survey shorter and easier to complete.

### 2. More Granular Brackets

On the opposite side, adding *more* brackets gives us better, more granular data. For example, with just 5 brackets you are limited to a smaller number of percentiles (at least, I think), which makes using e.g. box plots hard. 

So there's also an argument to be made for adding more brackets, especially for variables such as income that we might want to cross-reference with other datapoints. 

### 3. Equal Brackets

Consider two sets of brackets:

- 0-20k
- 20-40k
- 40-60k
- etc.

vs

- 0-10k
- 10-30k
- 30-100k
- 100-200k

In the second example the 30-100k bracket is 7 times larger than the 0-10k bracket, so that would probably skew the resulting data visualization in its favor, even with the exact same set of respondents incomes. 

Is this something we should be concerned about? On the other hand it's also clear that e.g. there is a big difference between making 10k and 50k a year, while 200k vs 250k isn't as meaningful, so a purely linear scale isn't always the best answer.

## Current Brackets

As reference, we currently have:

##### Yearly Income Brackets (in 10k USD)

- range_0_10
- range_10_30
- range_30_50
- range_50_100
- range_100_200
- range_more_than_200

##### Years of Experience Brackets

- range_less_than_1
- range_1_2
- range_2_5
- range_5_10 
- range_10_20
- range_more_than_20

##### Organization Size Brackets (number of people)

- range_1
- range_1_5
- range_5_10
- range_10_20
- range_20_50
- range_50_100
- range_100_1000
- range_more_than_1000

### Suggestions Welcome

Keeping all this in mind, I would love some suggestions for improving these brackets, if they need to be improved.

**Comment by ShaineRosewel** [comment]
Saw the detailed discussion from #177.

In my understanding, the reason for making this more granular is to be able to do midpoint coding, which I read from this [link](https://www.displayr.com/how-to-calculate-an-average-value-from-categorical-data/#:~:text=One%20rule%20is%20to%20halve,the%20number%20in%20the%20label.&text=By%20using%20these%20midpoints%20as,researcher%20can%20easily%20calculate%20averages.). Reason for wanting to do this is to be able to present results later on in terms of averages, not simply in terms of categories - and also because, we cannot ask exact values - privacy.

(Con) If we offer more granular options, it may take more time for the respondents to read - which means more effort. (Pro) But depending on how options are categorized, we can still combine some of them later on into fewer and bigger categories - leading us to @LeaVerou's suggested categorization.

Thing is, I think that once we come up with appropriate categorization, @LeaVerou's suggestion would be enough to show variance across categories, whenever there is. Hence, we won't need averages for results presentation later. Below was an example in her suggestion:

Just me!
Small (2 - 50)
Medium (51 - 200)
Large (201 - 1000)
Very Large (1000+)

With privacy being the main concern, to me, determining categories, based from some standard reference/s, that are known to have similarities within and variance across, so that we would not have to take averages, is fine

**Reply by SachaG** [reply]
It's nice to know the technical term (midpoint coding), thanks!

**Reply by ShaineRosewel** [reply]
Actually, so far, I only found that one reference. But in the context of measurement levels, converting categories into an exact number is similar to going from lower to a higher measurement level - which we are not supposed to do, ideally.

What is okay is converting from a higher to a lower level of measurement - say from exact values to categories (I think this has been mentioned in another discussion). By doing this, our data becomes less informative. It is also fine to collapse categories - more granular to less but larger ones. (we are not okay with the former due to privacy concerns.)

**Reply by SachaG** [reply]
> But in the context of measurement levels, converting categories into an exact number is similar to going from lower to a higher measurement level - which we are not supposed to do, ideally.

I think I'm ok with doing that as long as it's clearly explained. I don't really see a downside, practically speaking.

**Reply by ShaineRosewel** [reply]
Okay with this! Not the ideal approach given the data privacy stuff, but as mentioned in the reference there above,

_By using these midpoints as the categorical response values, the researcher can easily calculate averages. Granted, this average will only be an estimate or a “ballpark” value but is still extremely useful for the purpose of data analysis._

_It is also worth noting that **using more categories (and therefore smaller ranges) will result in a more accurate average as there will be less deviation from the actual value within these smaller ranges**._ :)

Best is we are aware of the downside so yes, have it clearly explained.

**Reply by SachaG** [reply]
With this in mind, if you have any suggestions for new brackets please let me know!

**Comment by SachaG** [comment]
OK so here's a proposal.

### Company Size

We leave it the same (8 brackets).

### Yearly Income

We add two brackets to get more granular and match Company Size:

- 0 to 20k
- 20 to 40k
- 40 to 60k
- 60 to 80k
- 80 to 100k
- 100 to 150k
- 150 to 200k
- 200k+

### Age

Number input. 

### Years of Experience

Number input? Still no 100% sure about this one.

**Reply by mikequiapos** [reply]
Here's a suggestion where I've considered 8 brackets per group. Using the data gathered from question, can enable segmenting the audience into appropriate groups and find the right audience. Your UI could be a simple drop down list-box.

**Company Size**
What is your Company Size?
Micro (1-10 employees)
Small (11-50 employees)
Medium (51-250 employees)
Large (251-1000 employees)
Enterprise (1001-5000 employees)
Corporate (5001-10,000 employees)
Global (10,001-50,000 employees)
Mega (50,001+ employees)

**Yearly Income**
What is your total income?
Less than $25,000
$25,000 - $50,000
$50,001 - $75,000
$75,001 - $100,000
$100,001 - $150,000
$150,001 - $200,000
$200,001 - $300,000
$300,001 and above

**Age**
What is your current age?
18-24 years old
25-34 years old
35-44 years old
45-54 years old
55-64 years old
65-74 years old
75-84 years old
85 years old and above

**Lastly, the Year of Experience**
How many years have you been using HTML?
Less than 1 year
1-3 years
4-6 years
7-10 years
11-15 years
16-20 years
21-25 years
26 years and above

**Reply by LeaVerou** [reply]
I think we need fewer brackets for company size, right now it's way too granular in the smaller company brackets. 
Age as number input is good. 
Years of experience, I'm also unsure, but leaning towards number input. Or maybe a slider showing e.g. "25+" at the last position.
Yearly income seems fine too.

**Reply by SachaG** [reply]
I think keeping 8 brackets for every place where we use brackets gives us a nice consistency, and also makes the chart more visually consistent.
